cmd/kind/load/docker-image/docker-image.go

```diff
@@ -88,18 +100,35 @@ func runE(flags *flagpole, cmd *cobra.Command, args []string) error {
 
 	// pick only the user selected nodes and ensure they exist
 	// the default is all nodes unless flags.Nodes is set
-	selectedNodes := nodes
+	candidateNodes := nodes
 	if len(flags.Nodes) > 0 {
-		selectedNodes := []clusternodes.Node{}
+		candidateNodes := []clusternodes.Node{}
 		for _, name := range flags.Nodes {
 			node, ok := nodesByName[name]
 			if !ok {
 				return errors.Errorf("unknown node: %s", name)
 			}
+			candidateNodes = append(candidateNodes, node)
+		}
+	}
+
+	// pick only the nodes that don't have the image
+	selectedNodes := []clusternodes.Node{}
+	for _, node := range candidateNodes {
+		cmd := node.Command("docker", "images", "-q", imageName)
```

Comment:
we just added the `docker.Images(imageName)` utility, yet in the case of nodes we still have to manually bake a command. can these two cases use the same utility?

Reply:
I was checking how it was done in the rest of the code and my understanding is that, when running commands on nodes, we always use this way.
However, if you point me to an example of using the ` docker.` utilities to run the command on nodes I'm happy to change it.

Reply:
we should add a `node.Images()` utility to list images available to kubernetes. this should use `ctr --namespace=k8s.io images list`

Reply:
I'm afraid that we have to use ` crictl -r /var/run/containerd/containerd.sock` instead of 
`ctr --namespace=k8s.io images` , I couldn't find same hashes or ids between `docker` and `ctr`

Reply:
using crictl and CRI in general where possible is ideal, mainly we just needed to not use `docker`, we should check the ctr output though at some point ourselves just as a sanity check, using crictl here is ideal.